### ISSUE TITLE
Ignore specific method calls in Metrics/AbcSize

### ISSUE THREAD

**Issue by CuddlyBunion341** [rubocop/rubocop#14180]
## The Problem

System specs often violate the `Metrics/AbcSize` rule due to many Capybara method calls, even when the logic is simple and well-structured. This leads to misleading complexity warnings.

Here’s a fictional example where the `AbcSize` rule could be intelligently ignored:

```rb
# frozen_string_literal: true

require "rails_helper"

RSpec.describe "Book Management", :js do
  it "can properly CRUD a book" do
    empty_book_index
    book_create
    book_show
    book_update_discard
    book_update_confirm
    present_book_index
  end

  def empty_book_index
    visit books_path
    expect(page).to have_content(I18n.t("books.index.title"))
    expect(page).to have_content(I18n.t("books.index.no_records"))
    click_on I18n.t("books.index.add_book")
  end

  def book_create
    visit new_book_path

    fill_in_book_fields

    click_on I18n.t("shared.save")
    expect(page).to have_content(I18n.t("books.create.success"))
  end

  def fill_in_book_fields
    fill_in Book.human_attribute_name(:isbn), with: "978-3-16-148410-0"
    fill_in Book.human_attribute_name(:title), with: "The Ruby Chronicles"
    fill_in Book.human_attribute_name(:author), with: "Ada Lovelace"
    fill_in Book.human_attribute_name(:published_on), with: "01.01.2020"
    select I18n.t("enums.book.genre.fiction"), from: Book.human_attribute_name(:genre)
    check Book.human_attribute_name(:bestseller)
  end

  def book_show
    expect(page).to have_content("The Ruby Chronicles")
    expect(page).to have_content("Ada Lovelace")
    expect(page).to have_content("978-3-16-148410-0")
  end

  def book_update_discard
    click_on I18n.t("books.edit")

    fill_in Book.human_attribute_name(:title), with: "The Rust Diaries"
    click_on I18n.t("shared.discard")

    expect(page).to have_no_content("The Rust Diaries")
  end

  def book_update_confirm
    click_on I18n.t("books.edit")

    fill_in Book.human_attribute_name(:title), with: "The Elixir Codex"
    fill_in Book.human_attribute_name(:author), with: "Grace Hopper"
    select I18n.t("enums.book.genre.non_fiction"), from: Book.human_attribute_name(:genre)

    click_on I18n.t("shared.save")
    expect(page).to have_content(I18n.t("books.update.success"))
    expect(page).to have_content("The Elixir Codex")
    expect(page).to have_content("Grace Hopper")
  end

  def present_book_index
    visit books_path
    expect(page).to have_content("The Elixir Codex")
    expect(page).to have_content("Grace Hopper")
    expect(page).to have_content("978-3-16-148410-0")
  end
end
```

This spec is clear and intentionally modular, but RuboCop flags methods like `fill_in_book_fields` for excessive complexity due to UI interactions.

## Proposed Solution

Allow configuring AbcSize to ignore specific method calls that don't contribute to actual logic complexity.

Example configuration (specific to capybara):

```yaml
Metrics/AbcSize:
  MethodsThatDontContributeToRule:
    -  fill_in
    - select
    - choose
    - check
    - uncheck
    - click_button
    - click_link
    - click_on
    - attach_file
    - visit
    - within
    - have_content
    - have_selector
    - find
```

## Alternatives Considered

* Excluding system specs from `AbcSize` entirely
* Disabling `AbcSize` inline (noisy and repetitive)
* Increasing the threshold globally (leads to abuse)

## Additional context

Capybara-heavy tests can look complex to RuboCop, but not to humans. Making `AbcSize` configurable for UI methods and others alike helps reduce noise and false positives while keeping the rule useful for actual complexity.

**Comment by Earlopain** [comment]
How about `AllowedPatterns`/`AllowedMethods`? I believe that should work (I have not tested this)

**Comment by CuddlyBunion341** [comment]
@Earlopain 
I have attempted to exclude the Methods 
AllowedMethods, but this seems to exclude the AbcSize of the method body, not the call of the method per se.

**Comment by cb341** [comment]
Could anyone look into it please?

**Comment by cb341** [comment]
Any interests in pursuing this issue further @Earlopain?
